File: 1713-Minimum-Operations-to-Make-a-Subsequence/main.go
package main

func minOperations(target []int, arr []int) int {

    targetIndex := make(map[int]int)
    for i, num := range target {
        targetIndex[num] = i
    }

    var targetIndices []int
    for _, num := range arr {
        if index, exists := targetIndex[num]; exists {
            targetIndices = append(targetIndices, index)
        }
    }

    lisLen := lenOfLis(targetIndices)
    return len(target) - lisLen
}

func lenOfLis(nums []int) int {
    lis := []int{}
    for _, num := range nums {
        pos := sort.SearchInts(lis, num)
        if pos < len(lis) {
            lis[pos] = num
        }else{
            lis = append(lis, num)
        }
    }

    return len(lis)
}

File: 1713-minimum-operations-to-make-a-subsequence/main.py
class Solution:
    def minOperations(self, target: List[int], arr: List[int]) -> int:
       
        target_index = {num: i for i, num in enumerate(target)}
        target_indices = [target_index[num] for num in arr if num in target_index]

        lis = []

        for index in target_indices:
            pos = bisect.bisect_left(lis, index)
            if pos == len(lis):
                lis.append(index)
            else:
                lis[pos] = index
        

        return len(target) - len(lis)

